src/component/EmailList/index.js

```diff
@@ -0,0 +1,134 @@
+import { useState } from "react";
+import EmailItem from "../EmailItem";
+import styles from "./EmailList.module.scss";
+import ChevronRightIcon from "@mui/icons-material/ChevronRight";
+import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
+import clsx from "clsx";
+import { useEffect } from "react";
+import { API_URL } from "../../constants";
+
+const token = localStorage.getItem("token");
+
+function EmailList() {
+  const [checkedItems, setCheckedItems] = useState([]);
+  const [emailList, setEmailList] = useState([]);
+  const [data, setData] = useState({});
+  const [page, setPage] = useState(1);
+  useEffect(() => {
+    fetch(
+      `${API_URL}/get-inbox?input=${JSON.stringify({
+        page: page,
+      })}`,
+      { headers: { Authorization: `Bearer ${localStorage.getItem("token")}` } }
+    )
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+      })
+      .then((data) => {
+        setEmailList(data.result.data.data);
+        setData(data.result.data);
+      });
+  }, [setEmailList]);
+  const handleCheckAll = () => {
+    if (emailList.length === checkedItems.length) {
+      setCheckedItems([]);
+      return;
+    }
+    setCheckedItems(emailList.map((item) => item.email.id));
+  };
+  const handleNext = () => {
+    if (emailList.length + (page - 1) * 50 < data.total) {
+      setPage(page + 1);
+    }
+  };
+  const handleBack = () => {
+    setPage(page - 1);
+  };
+  const handleStar = async (index) => {
+    const newEmail = { ...emailList[index] };
+    newEmail.starred = !emailList.starred;
+    const newEmailList = [...emailList];
+    newEmailList[index] = newEmail;
+    setEmailList(newEmailList);
+
+    fetch(`${API_URL}/toggle-star-email`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
+      },
+      body: JSON.stringify({
+        id: newEmail.email.id,
+      }),
+    })
+      .then((response) => response.json())
+      .then((result) => {});
+  };
+  return (
+    <>
+      <div className={styles.topbar}>
+        <div>
+          <input
+            type="checkbox"
+            checked={checkedItems.length === emailList.length}
+            onChange={handleCheckAll}
+          />
+        </div>
+        <div className={styles.dashbroadEmail}>
+          <span>
+            1-{emailList.length} of {data.total}
```

Comment:
Wrong!
It must be: 
```js
{(page - 1) * 50 + 1} - {(page - 1) * 50 + emailList.length} of {data.total}
```

For example, with page = 2, length = 10
-> 51 - 60 of 60